### PR TITLE
Nc news 7b

### PR DESCRIPTION
**Pull request by mikesealey**

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -196,13 +196,14 @@ describe("GET /api/articles/:article_id/comments", () => {
...
 // Ticket7
-describe.skip("POST /api/articles/:article_id/comments", () => {
+describe("POST /api/articles/:article_id/comments", () => {
     it("Should post a simple comment to an article that exists", () => {
         return request(app)
         .post("/api/articles/2/comments")
         .send({user_name: "butter_bridge", body: "TEST POST PLEASE IGNORE"})
         .expect(201)
         .then(({body}) => {
+            console.log(body)
```
Comment (Jodiech) [inline __tests__/api.test.js]: Please remove your console logs!

File: __tests__/api.test.js [hunk 2]
```diff
@@ -196,13 +196,14 @@ describe("GET /api/articles/:article_id/comments", () => {
...
             expect(body).toHaveProperty("article_id")
             expect(body).toHaveProperty("author")
             expect(body).toHaveProperty("body")
```
Comment (Jodiech) [inline __tests__/api.test.js]: I'd add another 201 test here that ignores additional properties such as:
**{
        username: "bainesface",
        body: "I loved this game too!",
        extraProperty: "extra things",
      }**
      
Please add a couple of tests for:

Status 400, missing required field(s), e.g. no username or body properties
Status 404, username does not exist

File: models/api.models.js [hunk 3]
```diff
@@ -69,13 +69,21 @@ exports.selectCommentsByArticleId = (id) => {
 
 // Ticket7
 exports.sendComment = (article_id, commentObject) => {
-    return db.query(`
+    // Re-use the selectArticleById function to make sure the article exists
+    return this.selectArticleById(article_id)
+    // If it doesn't exist, the returned rows will be empty
```
Comment (Jodiech) [inline models/api.models.js]: Please remove any comments! Keep your code nice and tidy!

**Review by Jodiech — COMMENTED** [review]
Hey 👋 Nice work on this ticket so far! I'd like you to add some tests to this, please see the comments I've left 😄 I'd like you to submit another PR once you've made these changes 👍

File: __tests__/api.test.js [hunk 4]
```diff
@@ -229,13 +229,50 @@ describe.skip("POST /api/articles/:article_id/comments", () => {
...
             expect(body.msg).toBe("Not found")
         })
     })
+    it("Should ignore additional properties when sent an object with more than just user_name and body", () => {
+        return request(app)
+        .post("/api/articles/2/comments")
+        .send({user_name: "butter_bridge", body: "This is the comment body", additionalProperty: "This is totally unnecessary third KVP that should have no effect on the query at all!"})
+        .expect(201)
+        .then(({body}) => {
+            expect(body).toHaveProperty("article_id")
```
Comment (lewispricey) [inline __tests__/api.test.js:265]: You could be specfic about the article id here since we're controlling what article the comment is for.

**Review by lewispricey — APPROVED** [review]
Great work on these extra tests Mike, thanks for implementing Jodie's feedback!

Good to merge when you're ready 🚀

**Pull request merged**